test/test_tree_3d_hypdiff.jl

```diff
@@ -13,19 +13,43 @@ EXAMPLES_DIR = pkgdir(Trixi, "examples", "tree_3d_dgsem")
       l2   = [0.001530331609036682, 0.011314177033289238, 0.011314177033289402, 0.011314177033289631],
       linf = [0.02263459033909354, 0.10139777904683545, 0.10139777904683545, 0.10139777904683545],
       initial_refinement_level=2)
+      # Ensure that we do not have excessive memory allocations 
+      # (e.g., from type instabilities) 
+      let 
+        t = sol.t[end] 
+        u_ode = sol.u[end] 
+        du_ode = similar(u_ode) 
+        @test (@allocated Trixi.rhs!(du_ode, u_ode, semi, t)) < 15000 
```

Comment:
Why are the numbers higher in this file?

Reply:
This is due to the usage of the custom integrators.
It appears that the Callbacks are not the only source of type instabilities, although #1684 greatly improved things.

Compare https://github.com/trixi-framework/Trixi.jl/actions/runs/6574426931/job/17859448565#step:7:3563
to https://github.com/trixi-framework/Trixi.jl/actions/runs/6613292878/job/17960862056?pr=1684#step:7:3586